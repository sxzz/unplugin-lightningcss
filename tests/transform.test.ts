import { resolve } from 'node:path'
import { describe } from 'vitest'
import { rollupBuild, testFixtures } from '@vue-macros/test-utils'
import css from 'rollup-plugin-css-only'
import LightningCSS from '../src/rollup'

describe('transform', async () => {
  await testFixtures(
    ['tests/fixtures/*.css'],
    (args, id) =>
      rollupBuild(id, [
        LightningCSS({
          options: {
            minify: true,
            targets: {
              ie: 11,
            },
          },
        }),
        css(),
      ]),
    { cwd: resolve(__dirname, '..'), promise: true },
  )
})
