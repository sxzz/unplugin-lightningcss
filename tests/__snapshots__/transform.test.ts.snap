// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > tests/fixtures/basic.css 1`] = `
"

.red{color:red}
"
`;

exports[`transform > tests/fixtures/import.css 1`] = `
"

@import "./basic.css";
"
`;

exports[`transform > tests/fixtures/nesting.css 1`] = `
"

.a .b{color:red}
"
`;
